gateways/firstdata/firstdata.go

```diff
@@ -0,0 +1,215 @@
+package firstdata
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+)
+
+const (
+	endpoint = "/payments"
+)
+
+type FirstdataClient struct {
+	host            string
+	apiKey          string
+	apiSecret       string
+	clientRequestID string
+	httpClient      *http.Client
+}
+
+func NewClient(env common.Environment, apiKey, apiSecret string) *FirstdataClient {
+	return NewWithHttpClient(env, apiKey, apiSecret, common.DefaultHttpClient())
+}
+
+func NewWithHttpClient(
+	env common.Environment,
+	apiKey,
+	apiSecret string,
+	httpClient *http.Client,
+) *FirstdataClient {
+	return &FirstdataClient{
+		host:       firstdataHost(env),
+		apiKey:     apiKey,
+		httpClient: httpClient,
+	}
+}
+
+func (client *FirstdataClient) primaryURL() string {
```

Comment:
nit: can we put comments at the top of all of these functions explaining what they will be used for